plugins/kafka/src/producer.cpp

```diff
@@ -0,0 +1,104 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "kafka/producer.hpp"
+
+#include <vast/detail/narrow.hpp>
+#include <vast/error.hpp>
+#include <vast/logger.hpp>
+
+#include <fmt/format.h>
+
+namespace vast::plugins::kafka {
+
+auto producer::make(configuration config) -> caf::expected<producer> {
+  producer result;
+  std::string error;
+  result.producer_.reset(RdKafka::Producer::create(config.conf_.get(), error));
+  if (!result.producer_)
+    return caf::make_error(ec::unspecified,
+                           fmt::format("failed to create producer: {}", error));
+  result.config_ = std::move(config);
+  return result;
+}
+
+auto producer::produce(std::string topic, std::span<const std::byte> bytes,
```

Comment:
Instead of the `const_cast` below, please change the API to take the proper type here already, and let the outermost context decide whether the `const_cast` is safe to do.

In this particular case, it is only safe if this operator has unique ownership of the chunk passed to it. It likely has, but if it does not then you would need to copy it first.

Reply:
I've provided reasoning for the `const_cast`.

This is an implementation detail and should _not_ leak out to the API.

Reply:
Does the reasoning in the comment work for you?

Reply:
Yes, thank you!